File: 완전탐색/지현/모음사전.py
# from itertools import permutations


# def solution(word):
#     alpha= ['A', 'E', 'I', 'O', 'U','AA', 'EE', 'II', 'OO', 'UU','AAA', 'EEE', 'III', 'OOO', 'UUU','AAAA', 'EEEE', 'IIII', 'OOOO', 'UUUU','AAAAA', 'EEEEE', 'IIIII', 'OOOOO', 'UUUUU']

#     permutation=[]
#     for i in permutations(alpha,5):
#         if len(''.join(i))<=5:
#             permutation.append(''.join(i))
#     # print (permutation)

#     answer = 0
#     permutation.sort()
#     permutation.index(word)
#     return answer
ㅎ
from itertools import product   #중복순열로 푼다.
def solution(word):
    permutation=[]
    for j in range(1,6):   #길이 1~5개 있으니 반복
        for i in product(['A', 'E', 'I', 'O', 'U'], repeat=j):  #반복개수 지정하고 중복순열 구하기
            permutation.append(''.join(i))  # ''.join(list)로 list->string으로 변환
    # print(permutation)
    answer=0
    permutation.sort()  #사전이니까 sort()
    answer=permutation.index(word)+1
    return answer

File: 완전탐색/[희진]완전탐색/모음사전.js
function solution(word) {
  let charArr = ["A", "E", "I", "O", "U"];
  let t = [];
  var answer = 0;

  // 깊이 우선 탐색 (DFS)을 통해 길이가 5 이하인 단어를 생성하여 배열에 저장
  const dfs = (cha) => {
    if (cha.length > 5) {
      return;
    }
    t.push(cha); // 현재 단어를 배열에 추가

    // 각 알파벳으로 분기하여 다음 단계의 DFS 호출
    for (let i = 0; i < 5; i++) {
      dfs(cha + charArr[i]);
    }
  };

  // 각 알파벳으로 시작하는 단어를 생성하기 위해 DFS 호출
  for (let a of charArr) {
    dfs(a);
  }

  // 생성된 단어들을 내림차순으로 정렬
  t.sort((a, b) => b - a);

  // 주어진 단어가 정렬된 배열에서 몇 번째 위치하는지 계산
  answer = 1 + t.indexOf(word);

  return answer;
}
